### PR TITLE
fix: low: underflow instead of error

### PR DESCRIPTION
**Pull request by drgorillamd**
Fix the flow:
- no target set
- balance < allowance
- try to use an allowance > balance

Expected: revert with INADEQUATE_WITHDRAW_AMOUNT

Observed: revert on underflow (balance-withdraw) or on INADEQUATE_PAYMENT_TERMINAL_STORE_BALANCE if a target is set

Fix: refactor (test for amount > balance)

https://discord.com/channels/775859454780244028/911325354218631208/946702526537949204

### CODE REVIEW

**Review by mejango — APPROVED** [review]
lgtm

File: contracts/JBETHPaymentTerminalStore.sol [hunk 1]
```diff
@@ -474,6 +474,11 @@ contract JBETHPaymentTerminalStore {
       ? _amount
       : PRBMathUD60x18.div(_amount, prices.priceFor(_currency, JBCurrencies.ETH));
 
+    // The amount being withdrawn must be at least as much as was expected.
+    if (_minReturnedWei > withdrawnAmount || balanceOf[_projectId] < withdrawnAmount) {
+      revert INADEQUATE_WITHDRAW_AMOUNT();
```
Comment (dhyon) [inline contracts/JBETHPaymentTerminalStore.sol:479]: hmm I think this error message only applies to the `_minReturnedWei > withdrawnAmount`  case

I think `INADEQUATE_PAYMENT_TERMINAL_STORE_BALANCE()` is more appropriate for the new case you've added

File: contracts/JBETHPaymentTerminalStore.sol [hunk 2]
```diff
@@ -859,4 +859,4 @@ contract JBETHPaymentTerminalStore {
         ? 0
         : _ethBalanceOf - _ethDistributionLimitRemaining;
   }
-}
+}
```
Comment (dhyon) [inline contracts/JBETHPaymentTerminalStore.sol:862]: newline plox

**Review by dhyon — COMMENTED** [review]
system tests coming through!

File: contracts/JBETHPaymentTerminalStore.sol (hunk 1 above)
Comment (drgorillamd) [reply contracts/JBETHPaymentTerminalStore.sol:479]: Is it the withdrawAmount or the terminal balance fault :D ?

Comment (mejango) [reply contracts/JBETHPaymentTerminalStore.sol:479]: maybe two separate checks?

Comment (drgorillamd) [reply contracts/JBETHPaymentTerminalStore.sol:484]: done!

Comment (dhyon) [reply contracts/JBETHPaymentTerminalStore.sol:484]: ah just saw this, thanks for handling it

**Pull request merged**